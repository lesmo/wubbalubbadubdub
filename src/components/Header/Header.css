.logoContainer {
  text-align: center;
}
.logoContainer img {
  max-width: 70%;
  max-height: 180px;
}
