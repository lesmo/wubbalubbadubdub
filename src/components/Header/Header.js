import React from 'react';
import withStyles from 'isomorphic-style-loader/withStyles';
import Grid from '@material-ui/core/Grid';

import Search from '../Search';
import LanguageSwitcher from '../LanguageSwitcher';

import s from './Header.css';
import logo from './logo.png';

class Header extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Grid container direction="row" justify="center" alignItems="center">
          <Grid item xs={12} className={s.logoContainer}>
            <img src={logo} />
          </Grid>
          <Grid item>
            <Search />
          </Grid>
        </Grid>
        <LanguageSwitcher />
      </React.Fragment>
    );
  }
}

export default withStyles(s)(Header);
