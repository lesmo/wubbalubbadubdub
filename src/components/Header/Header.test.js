/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { IntlProvider } from 'react-intl';

import createApolloClient from '../../core/createApolloClient';

import App from '../App';
import Header from './Header';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Header', () => {
  ['en-US', 'es-MX'].map(lang => {
    test(`render correcto de header (${lang})`, () => {
      const initialState = {
        runtime: {
          availableLocales: [lang],
        },
        intl: {
          locale: lang,
        },
      };

      const store = mockStore(initialState);
      const client = createApolloClient();
      const { intl } = new IntlProvider({
        initialNow: new Date(15e11),
        defaultLocale: lang,
        locale: lang,
        messages: {},
      }).getChildContext();
  
      const wrapper = renderer
        .create(
          <App
            context={{
              insertCss: () => { },
              fetch: () => { },
              pathname: '',
              intl,
              store,
              client,
            }}
          >
            <Header />
          </App>,
        )
        .toJSON();
  
      expect(wrapper).toMatchSnapshot();
    });
  });
});
