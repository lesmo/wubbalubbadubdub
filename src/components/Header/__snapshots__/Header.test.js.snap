// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correcto de header (en-US) 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root logoContainer MuiGrid-item MuiGrid-grid-xs-12"
    >
      <img
        src="logo.png"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <form
        method="get"
        onSubmit={[Function]}
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-104 MuiPaper-rounded"
        >
          <div
            className="MuiInputBase-root makeStyles-input-105"
            onClick={[Function]}
          >
            <input
              aria-describedby={undefined}
              aria-invalid={undefined}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInputBase-input"
              defaultValue={undefined}
              disabled={undefined}
              id={undefined}
              name={undefined}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder="Search the Universe"
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="text"
              value={undefined}
            />
          </div>
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-106"
            disabled={false}
            onBlur={[Function]}
            onClick={undefined}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                color={undefined}
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>,
  <div
    className="root"
  >
    <div
      className="MuiButtonGroup-root"
      role="group"
    >
      <a
        aria-disabled={false}
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiTypography-colorPrimary"
        href="?lang=en-US"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          EN
        </span>
      </a>
    </div>
  </div>,
]
`;

exports[`Header render correcto de header (es-MX) 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root logoContainer MuiGrid-item MuiGrid-grid-xs-12"
    >
      <img
        src="logo.png"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <form
        method="get"
        onSubmit={[Function]}
      >
        <div
          className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-104 MuiPaper-rounded"
        >
          <div
            className="MuiInputBase-root makeStyles-input-105"
            onClick={[Function]}
          >
            <input
              aria-describedby={undefined}
              aria-invalid={undefined}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInputBase-input"
              defaultValue={undefined}
              disabled={undefined}
              id={undefined}
              name={undefined}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder="Search the Universe"
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="text"
              value={undefined}
            />
          </div>
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-106"
            disabled={false}
            onBlur={[Function]}
            onClick={undefined}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                color={undefined}
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>,
  <div
    className="root"
  >
    <div
      className="MuiButtonGroup-root"
      role="group"
    >
      <a
        aria-disabled={false}
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiTypography-colorPrimary"
        href="?lang=es-MX"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          ES
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>,
]
`;
